addon/errors.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * Checks if the given response is a FetchError
+ */
+export function isFetchError(response: Response): boolean {
+  return response.ok === false
+}
+
+/**
+ * Checks if the given response represents an unauthorized request error
+ */
+export function isUnauthorizedError(response: Response): boolean {
+  if (isFetchError(response)) {
+    return response.status === 401;
+  }
+
+  return false
+}
+
+/**
+ * Checks if the given response represents a forbidden request error
+ */
+export function isForbiddenError(response: Response): boolean {
+  if (isFetchError(response)) {
+    return response.status === 403;
+  }
+
+  return false
+}
+
+/**
+ * Checks if the given response represents an invalid request error
+ */
+export function isInvalidError(response: Response): boolean {
+  if (isFetchError(response)) {
+    return response.status === 422;
+  }
+
+  return false
+}
+
+/**
+ * Checks if the given response represents a bad request error
+ */
+export function isBadRequestError(response: Response): boolean {
+  if (isFetchError(response)) {
+    return response.status === 400;
+  }
+
+  return false
+}
+
+/**
+ * Checks if the given response represents a "not found" error
+ */
+export function isNotFoundError(response: Response): boolean {
+  if (isFetchError(response)) {
+    return response.status === 404;
+  }
+
+  return false
+}
+
+/**
+ * Checks if the given response represents a "gone" error
+ */
+export function isGoneError(response: Response): boolean {
+  if (isFetchError(response)) {
+    return response.status === 410;
+  }
+
+  return false
+}
+
+/**
+ * Checks if the given response represents an "abort" error
+ */
+export function isAbortError(response: Response): boolean {
+  if (isFetchError(response)) {
+    return response.status === 0;
+  }
+
+  return false
+}
+
+/**
+ * Checks if the given response represents a conflict error
+ */
+export function isConflictError(response: Response): boolean {
+  if (isFetchError(response)) {
+    return response.status === 409;
+  }
+
+  return false
+}
+
+/**
+ * Checks if the given response represents a server error
+ */
+export function isServerError(response: Response): boolean {
+  if (isFetchError(response)) {
+    return response.status >= 500 && response.status < 600;
+  }
+
+  return false
+}
+
+/**
+ * Checks if the given status code represents a successful request
+ */
+export function isSuccess(response: Response): boolean {
```

Comment:
ember-ajax's `isSuccess` is for checking status code, do you think we should continue to make it checking status or response object?

Reply:
Similar to `isFetchError`, we probably don't need `isSuccess`?

```js
fetch().then(() => {
// is success
}).catch(() => {
// is error
})
```

Reply:
Yea agree, it doesn't really provide that much value since it's just one liner `r => r.ok`...

Maybe rename `isXXXError` to `isXXXResponse`? And `isNetworkError` and `isAbortError
```
fetch().then((response) => {
    if (response.ok) {
        // is success
    } else  {
        // is xxx response
    }
}).catch((error) => {
        // is xxx error
})
```